openverse_catalog/dags/providers/provider_api_scripts/science_museum.py

```diff
@@ -45,198 +27,202 @@
     (1990, 2020),
 ]
 
-# global variable to keep track of records pulled
-RECORD_IDS = []
-
 
-def main():
-    logger.info("Begin: Science Museum script")
-    for year_range in YEAR_RANGE:
-        logger.info(f"Running for years {year_range}")
-        from_year, to_year = year_range
-        image_count = _page_records(from_year=from_year, to_year=to_year)
-        logger.info(f"Images pulled till now {image_count}")
-    image_count = image_store.commit()
-    logger.info(f"Total images pulled {image_count}")
-
-
-def _page_records(from_year, to_year):
-    image_count = 0
-    page_number = 0
-    condition = True
-    while condition:
-        query_param = _get_query_param(
-            page_number=page_number, from_year=from_year, to_year=to_year
-        )
-        batch_data = _get_batch_objects(query_param=query_param)
-        if type(batch_data) == list:
-            if len(batch_data) > 0:
-                image_count = _handle_object_data(batch_data)
-                page_number += 1
-            else:
-                condition = False
+class ScienceMuseumDataIngester(ProviderDataIngester):
+    providers = {"image": prov.SCIENCE_DEFAULT_PROVIDER}
+    endpoint = "https://collection.sciencemuseumgroup.org.uk/search/"
+    batch_limit = 1000
+    delay = 5
+    headers = {"Accept": "application/json"}
+    RECORD_IDS = []  # class variable to keep track of records pulled
+    # I set this to 10 so that it stops right after the first year_range is processed
+
+    def ingest_records(self, **kwargs):
+        for year_range in YEAR_RANGES:
+            super().ingest_records(year_range=year_range)
+
+    def get_next_query_params(self, old_query_params, **kwargs):
+        from_ = kwargs["year_range"][0]
+        to_ = kwargs["year_range"][1]
```

Comment:
This might be clearer as:
```suggestion
        from_, to_ = kwargs["year_range"]
```